### ISSUE TITLE
Fix relative path in 'installpath' option

### ISSUE THREAD

**Issue by MisterFruits** [easybuilders/easybuild-framework#2467]
Currently if you try to build a software with EB giving ``--installpath=a/relative/path`` it will work until some point (I managed to install some zlib-1.2.10) but once you reach a software that have dependencies it will fail with a non clear message:

Given a deptest.eb that declared ``dependencies = [('zlib', '1.2.10')]``

```
$ eb --installpath=build/eb zlib-1.2.10.eb deptest.eb
[...]
== zlib/1.2.10 is already installed (module found), skiping
== resolving dependencies ...
== processing Easybuild easyconfig [...]/deptest.eb
[...]
== sanity checking
== FAILED: Installation ended unsuccessfully (build directory: [...]/build/[...]: build failed
(first 300 chars): zlib(3):ERROR:105: Unable to locate a module for 'zlib/1.2.10'
[...]
ERROR: Build of deptest.eb faild [...]
```

Giving ``--installpaht=$PWD/build/eb`` works.

To fix that, I would propose consolidating the relative path to an abolute path at config parsing time.

If you're not in favour of allowing relative path one could also detect any relative path given to ``eb --installpath`` and trow a related error message and stop before doing anything. Avoiding confusion. 

I would push in favour of the option allowing relative path because:
 - It would give more flexibility to EB (like while testing in an unknow CI environment, it would the necessity of generating the configuration file based on the current directory)
 - People who don't like that can still use absolut paths

What do you think ? I could contribute in the code to adress this request

**Comment by boegel** [comment]
I agree that we should allow relative paths and transform them into an absolute path based on the current working directory.

This problem is probably not limited to just `--installpath` though, so we should try and fix it in a general sense?

May be related to https://github.com/easybuilders/easybuild-framework/issues/2372
